pkg/extension/sumologicextension/credentialsstore_localfs.go

```diff
@@ -0,0 +1,143 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sumologicextension
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+
+	"go.uber.org/zap"
+)
+
+// localFsCredentialsStore implements CredentialsStore interface and can be used
+// to store and retrieve collector credentials from local file system.
+//
+// Files are stored locally in collectorCredentialsDirectory.
+type localFsCredentialsStore struct {
+	collectorCredentialsDirectory string
+	logger                        *zap.Logger
+}
+
+// Check checks if collector credentials can be found under a name being a hash
+// of provided key inside collectorCredentialsDirectory.
+func (cr localFsCredentialsStore) Check(key string) bool {
+	filenameHash, err := hash(key)
+	if err != nil {
+		return false
+	}
+	path := path.Join(cr.collectorCredentialsDirectory, filenameHash)
+	if _, err := os.Stat(path); err != nil {
+		return false
+	}
+	return true
+}
+
+// Get retrieves collector credentials stored in local file system and then
+// decrypts it using a hash of provided key.
+func (cr localFsCredentialsStore) Get(key string) (CollectorCredentials, error) {
+	filenameHash, err := hash(key)
+	if err != nil {
+		return CollectorCredentials{}, err
+	}
+
+	path := path.Join(cr.collectorCredentialsDirectory, filenameHash)
+	creds, err := os.Open(path)
+	if err != nil {
+		return CollectorCredentials{}, err
+	}
+
```

Comment:
It is only suggestion but I don't like this empty line, I prefer when defer is as close as possible to `os.Open` and here is added to another block of code.

Reply:
I agree and I also prefer it this way, thanks!

Reply:
Changed